### PR TITLE
447 encounter details 1

### LINKED ISSUE #447: Encounter details view
- [x] encounter info view (in the app bar) (1) 
- [x] create "bubble tab" interface, each bubble is a tab (implemented as a JSONForms tab control) (2) 
- [x] each tab is displayed as a modal (3)

(1) encounter info view
Image

(2) bubble tab interface
Image

(3) tab as modal
Image

### PR DESCRIPTION
**Pull request by mark-prins**
Closes #447 (partial)

The first steps for the encounter detail view. This PR will
- create the Encounters section, under Patients
- add a list view and detail view for encounters
- add a toolbar to the detail view which has editable date/time fields
- changes the patient encounter list view to redirect to the new detail view rather than the modal ( this appears to be the intention )

Questions:
- I've used the `id` of the encounter which isn't very elegant but is better than the `name`. thoughts?
- I've added the patient to the encounter query, as the patient is needed to be displayed in the list and detail views. this ok?
- Have used a debounce callback to update the fields, similar to what we've done elsewhere. there doesn't seem to be a place to have a save button, and this seems best for me. thoughts?

Todo:
- will update the sort options in the list view.. not all of them will work atm!

## Tests
- [ ] can see the Encounters menu link
- [ ] takes you to the list view
- [ ] results are shown
- [ ] clicking on a row takes you to the detail view
- [ ] you can edit each of the time/date fields
- [ ] the patient > encounter list view also redirects you to the Encounter / Detail View when clicked

### CODE REVIEW

File: client/packages/common/src/authentication/api/hooks/useUserDetails.ts [hunk 1]
```diff
@@ -1,15 +1,9 @@
...
 
 export const useUserDetails = () => {
   const api = useAuthApi();
-  return useMutation<
-    UserNode | undefined,
-    unknown,
-    string | undefined,
-    unknown
-  >(api.get.me);
+  return useMutation(api.get.me);
```
Comment (mark-prins) [inline client/packages/common/src/authentication/api/hooks/useUserDetails.ts:6]: 🤷  it was complaining after the merge from develop

File: client/packages/common/src/ui/components/inputs/TimePickers/TimePickerInput/TimePickerInput.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React, { FC } from 'react';
```
Comment (mark-prins) [inline client/packages/common/src/ui/components/inputs/TimePickers/TimePickerInput/TimePickerInput.tsx:1]: will need to add a story for this too..

File: client/packages/system/src/Encounter/DetailView/DetailView.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  const t = useTranslation('patients');
+  const { data: encounter, isLoading } = useEncounter.document.get();
+  const navigate = useNavigate();
+
+  const handleSave = useEncounter.document.upsert(
+    encounter?.patient.id ?? '',
+    '',
+    encounter?.document.documentRegistry?.documentType ?? ''
+  );
+  const { JsonForm /* saveData, isDirty, validationError */ } = useJsonForms(
```
Comment (mark-prins) [inline client/packages/system/src/Encounter/DetailView/DetailView.tsx:26]: this is a little temporary.. have just pulled the json data in for now - but will change to the bubble tabs

File: client/packages/system/src/Encounter/DetailView/DetailView.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+export const DetailView: FC = () => {
+  const t = useTranslation('patients');
+  const { data: encounter, isLoading } = useEncounter.document.get();
+  const navigate = useNavigate();
+
+  const handleSave = useEncounter.document.upsert(
+    encounter?.patient.id ?? '',
+    encounter?.program ?? '',
+    encounter?.type ?? ''
```
Comment (clemens-msupply) [inline client/packages/system/src/Encounter/DetailView/DetailView.tsx:24]: I used the correct value here...

File: client/packages/system/src/Encounter/DetailView/DetailView.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  const updateEncounter = useDebounceCallback(
+    (patch: Partial<EncounterFragment>) =>
+      handleSave?.(
+        { ...encounter?.document.data, ...patch },
+        encounter?.document?.documentRegistry?.formSchemaId ?? '',
+        encounter?.document.id
+      ),
+    [encounter],
+    1000
+  );
```
Comment (clemens-msupply) [inline client/packages/system/src/Encounter/DetailView/DetailView.tsx:43]: We would  need a save button at some point, similar to patient view. For example, a lab report could be added to an existing encounter.

File: client/packages/system/src/Encounter/ListView/ListView.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+export const EncounterListView: FC = () => (
+  <TableProvider
+    createStore={createTableStore}
+    queryParamsStore={createQueryParamsStore<EncounterFragmentWithId>({
+      initialSortBy: { key: 'startDatetime' },
+    })}
+  >
+    <EncounterListComponent />
+  </TableProvider>
```
Comment (clemens-msupply) [inline client/packages/system/src/Encounter/ListView/ListView.tsx:103]: Reuse existing list view if the differences don't get too big?

**Review by clemens-msupply — APPROVED** [review]
Looks good so far!

**Comment by clemens-msupply** [comment]
> I've used the id of the encounter which isn't very elegant but is better than the name. thoughts?

Good question, maybe the start_date since it is a required field...

> I've added the patient to the encounter query, as the patient is needed to be displayed in the list and detail views. this ok?

sounds good

> Have used a debounce callback to update the fields, similar to what we've done elsewhere. there doesn't seem to be a place to have a save button, and this seems best for me. thoughts?

think we need a safe button similar to the patient details view

**Pull request merged**